exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.testkit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.Lists.asList;
+import static java.util.Collections.singletonList;
+
+import com.exonum.binding.blockchain.Block;
+import com.exonum.binding.proxy.NativeHandle;
+import com.exonum.binding.runtime.ReflectiveModuleSupplier;
+import com.exonum.binding.service.BlockCommittedEvent;
+import com.exonum.binding.service.Service;
+import com.exonum.binding.service.ServiceModule;
+import com.exonum.binding.service.adapters.UserServiceAdapter;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
+ * TestKit for testing blockchain services. It offers simple network configuration emulation
+ * (with no real network setup). Although it is possible to add several validator nodes to this
+ * network, only one node will create the service instances and will execute their operations
+ * (e.g., {@link Service#afterCommit(BlockCommittedEvent)} method logic).
+ *
+ * <p>When TestKit is created, Exonum blockchain instance with given services is initialized and
```

Comment:
@vitvakatu, Would you clarify please what is the exact order? Do we initialize after
genesis block? When do we create public API handlers?

Reply:
The order is `initialize -> genesis block commit -> create handlers`. Service::initialize is called on genesis block creation, so actual it looks more like `initialize and genesis block commit -> create handlers`

Reply:
Thanks, it seems this description needs to be updated to reflect the actual order of operations.